### PR TITLE
BugFix: uk_sched_set_default now updates the uk_sched_head

### LINKED ISSUE #381: Bug in scheduler
### Describe the bug

A bug is fixed in int uk_sched_set_default(struct uk_sched *s) that was not updating the uk_sched_head pointer after moving the default schedule pointer to top of the queue.

struct uk_sched *s = uk_sched_head->next;
uk_sched_set_default(s); //where, s is in the list
struct uk_sched *t = uk_sched_get_default();
assert(t == s); //Failed due to this bug

### Fixing the bug

A fix is provided in this [pull](https://github.com/unikraft/unikraft/pull/380) request.

### PR DESCRIPTION
**Pull request by farif**
### Prerequisite checklist

 - [x] Read the [contribution guidelines](https://docs.unikraft.org/contribute.html) regarding submitting new changes to the project;
 - [x]  Tested your changes against relevant architectures and platforms;
 - [x]  Ran the [`checkpatch.pl`](https://github.com/unikraft/unikraft/blob/staging/support/scripts/checkpatch.pl) on your commit series before opening this PR;
 - [x]  Updated relevant documentation.

### Base target

 - Architecture(s): [e.g. `x86_64` or N/A]
 - Platform(s): [e.g. `kvm`, `xen` or N/A]
 - Application(s): [e.g. `app-python3` or N/A]

### Description of changes
A bug is fixed in `int uk_sched_set_default(struct uk_sched *s)` that was not  updating the `uk_sched_head` pointer after moving the default schedule pointer to top of the queue.
```
struct uk_sched *s = uk_sched_head->next;
uk_sched_set_default(s); //where, s is in the list
struct uk_sched *t = uk_sched_get_default();
assert(t == s); //Failed due to this bug
```

### CODE REVIEW

File: lib/uksched/sched.c
Comment (nderjung) [inline lib/uksched/sched.c]:
```diff
- 		//update the default scheduler to head
+ 		// Update the default scheduler to head
```

**Review by vladandrew — APPROVED** [review]
Hi Fareed,

Thank you for sending this pull request. Yes, I am able to reproduce this bug. Moving a registered scheduler at the top with `uk_sched_set_default` does not work properly. This patch fixes the problem. Only one comment, could you rename the commit to follow the Unikraft guidelines: `lib/uksched: uk_sched_set_default now updates the uk_sched_head` and add the sign off before the patch being approved.

Reviewed-by: Vlad-Andrei Badoiu <vlad_andrei.badoiu@upb.ro>

**Review by nderjung — CHANGES_REQUESTED** [review]
Hi @farif, thanks for the PR!

There are a few things which need to be addressed in this PR before I can mark it as approved:

 * The checkpatch has failed.  Please can you run it before submitting a PR; it's asking you to sign-off your commits (`git commit -s`).  I can also see that the commit message needs a few changes in addition to the sign-off:
   - Please can add mandatory path prefix, i.e. `lib/uksched: Make uk_sched_set_default update uk_sched_head`.
   - Please provide a longer description in the commit message explaining that this is a bug fix, what was the previous functionality and what the intention/result of the fix is and its implication.
 * A small adjustment is needed in the inline comment (see below).
 
Let's also wait for @vladandrew to test the changes, thanks for providing the test code to make this check too.  Maybe something for the future, this code would make for a nice internal test for the scheduler in conjunction with `uktest`.

Thanks!

**Comment by nderjung** [comment]
Thanks for the quick review @vladandrew 🔥 

Please can you just fix the commit message and comment @farif and then it's good to go.  Simply force push to the branch. 

Please in the future can you also use a different branch name when you request to merge into unikraft, it makes checking it out easier, e.g. a branch name called `farif/fix-381`.

File: lib/uksched/sched.c
Comment (farif) [reply lib/uksched/sched.c]: Thank you guiding me. I made all the requested change to the comment and sign-off the commit.

**Review by farif — COMMENTED** [review]
I made the request change and sign-off my commit.
Thank you for guiding me.

**Comment by nderjung** [comment]
Hey @farif, thanks for making the changes! Please can you squash the two commits into one and use the title and contents of the second, then just force push to your branch.

**Comment by farif** [comment]
Done:) Have I performed the merging of commits in a correct fashion?

**Comment by nderjung** [comment]
Hi @farif, unfortunately no.  We are expecting 1 commit rather than 4. 

Try this:
```bash
cd path/to/unikraft
git reset HEAD~4
git add lib/uksched/sched.c 
git commit -s -m "lib/uksched: Make uk_sched_set_default update uk_sched_head"
git push -u --force origin staging
```

**Comment by farif** [comment]
Thank you. Done.

**Review by vladandrew — APPROVED** [review]

**Comment by nderjung** [comment]
Perfect, thanks!

Approved-by: Alexander Jung <a.jung@lancs.ac.uk>

**Review by nderjung — APPROVED** [review]
Perfect, thanks!

Approved-by: Alexander Jung <a.jung@lancs.ac.uk>

**Pull request closed without merging**
